bindings/c/foundationdb/fdb_c.h

```diff
@@ -63,6 +63,8 @@
 extern "C" {
 #endif
 
+#ifndef FDB_API_OBJECTS
```

Comment:
This was actually one of the things I had trouble deciding on which was the best option, as all seem to have design tradeoffs. I'll list them out here so we can just agree upon one.

1. Include `fdb_c.h` in `fdb_c_internal.h`
2. Include `fdb_c_internal.h` in `fdb_c_h`
3. Separate the two and include them explicitly in files that look for them

The issue with (1) is that in order to include `fdb_c.h`, we must first `#define FDB_API_VERSION` which isn't a big deal, but could cause confusion with include ordering, and would require the version defined to be continually updated otherwise we risk overwriting other definitions once the api version advances. (2) and (3) share the issue where type re-definitions are not supported prior to the 2011 C standard (meaning it won't compile).

Reply:
Is there a workable option that involves putting the shared definitions in a third header that is included in both?

Reply:
@sfc-gh-ajbeamon that sounds like a reasonable option. I suppose it would only involve moving what I have surrounded by `#ifndef FDB_API_OBJECTS` into its own file then? Should we aim to extract more than that? There are definitely quite a few typedefs in `fdb_c.`, but the logic grows a bit more complicated and starts depending on `FDB_API_VERSION`

Reply:
The simplest approach is to move the minimal set of functionality that you need, and we could worry about other stuff later.

Out of curiosity, how are you dealing with versioning of the internal APIs? Are these going to use the API version mechanism?

Reply:
Currently, there isn't any plan to use the API version mechanism for the internal APIs. The main motivation for the separation of the file was so that it was clear these would not be part of the public interface. Do you have any comments on whether this should be changed?

Reply:
It doesn't necessarily need to be versioned with API versioning, though we should have a plan for how this API could be allowed to evolve over time if we choose to do something else. We could use API versioning even if they weren't part of the public interface, but we also have more flexibility to take other paths.